plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialSwitchFace.java

```diff
@@ -3,99 +3,94 @@
 import com.denizenscript.denizen.objects.MaterialTag;
 import com.denizenscript.denizencore.objects.core.ElementTag;
 import com.denizenscript.denizencore.objects.Mechanism;
-import com.denizenscript.denizencore.objects.ObjectTag;
-import com.denizenscript.denizencore.objects.properties.Property;
-import com.denizenscript.denizencore.objects.properties.PropertyParser;
 import org.bukkit.block.BlockFace;
+import org.bukkit.block.data.BlockData;
 import org.bukkit.block.data.Directional;
 import org.bukkit.block.data.FaceAttachable;
+import org.bukkit.block.data.type.Bell;
 
-public class MaterialSwitchFace implements Property {
+public class MaterialSwitchFace extends MaterialProperty<ElementTag> {
 
-    public static boolean describes(ObjectTag material) {
-        return material instanceof MaterialTag
-                && ((MaterialTag) material).hasModernData()
-                && ((MaterialTag) material).getModernData() instanceof FaceAttachable;
+    // <--[property]
+    // @object MaterialTag
+    // @name switch_face

```

Comment:
probably time to rename this property to be more generic, given that it's very much not switch-specific